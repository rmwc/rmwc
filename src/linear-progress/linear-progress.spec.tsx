import React from 'react';
import { render } from '@testing-library/react';
import { LinearProgress } from './';

describe('LinearProgress', () => {
  it('renders', () => {
    const { asFragment, container } = render(<LinearProgress progress={0.5} />);

    expect(container.firstChild).toHaveClass('mdc-linear-progress');
    expect(asFragment()).toMatchSnapshot();
  });

  it('can buffer', () => {
    const { asFragment } = render(<LinearProgress buffer={0.8} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can be closed', () => {
    const { asFragment } = render(<LinearProgress closed />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can be indeterminate', () => {
    const { container } = render(<LinearProgress />);
    expect(container.firstChild).not.toHaveClass(
      'mdc-linear-progress--determinate'
    );
  });

  it('can be reversed', () => {
    const { container } = render(<LinearProgress reversed />);
    expect(container.firstChild).toHaveClass('mdc-linear-progress--reversed');
  });

  it('can have custom classnames', () => {
    const { container } = render(
      <LinearProgress className={'my-custom-classname'} />
    );
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });
});
