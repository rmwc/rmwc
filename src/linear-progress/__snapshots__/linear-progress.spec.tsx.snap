// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinearProgress can be closed 1`] = `
<DocumentFragment>
  <nav
    aria-label="Progress Bar"
    aria-valuemax="1"
    aria-valuemin="0"
    class="mdc-linear-progress mdc-linear-progress--closed mdc-linear-progress--indeterminate mdc-linear-progress--closed"
    role="progressbar"
  >
    <div
      class="mdc-linear-progress__buffer"
    >
      <div
        class="mdc-linear-progress__buffer-bar"
        style="flex-basis: 100%;"
      />
      <div
        class="mdc-linear-progress__buffer-dots"
      />
    </div>
    <div
      class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style="transform: scaleX(1);"
    >
      <div
        class="mdc-linear-progress__bar-inner"
      />
    </div>
    <div
      class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <div
        class="mdc-linear-progress__bar-inner"
      />
    </div>
  </nav>
</DocumentFragment>
`;

exports[`LinearProgress can buffer 1`] = `
<DocumentFragment>
  <nav
    aria-label="Progress Bar"
    aria-valuemax="1"
    aria-valuemin="0"
    class="mdc-linear-progress mdc-linear-progress--indeterminate"
    role="progressbar"
  >
    <div
      class="mdc-linear-progress__buffer"
    >
      <div
        class="mdc-linear-progress__buffer-bar"
        style="flex-basis: 100%;"
      />
      <div
        class="mdc-linear-progress__buffer-dots"
      />
    </div>
    <div
      class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style="transform: scaleX(1);"
    >
      <div
        class="mdc-linear-progress__bar-inner"
      />
    </div>
    <div
      class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <div
        class="mdc-linear-progress__bar-inner"
      />
    </div>
  </nav>
</DocumentFragment>
`;

exports[`LinearProgress renders 1`] = `
<DocumentFragment>
  <nav
    aria-label="Progress Bar"
    aria-valuemax="1"
    aria-valuemin="0"
    aria-valuenow="0.5"
    class="mdc-linear-progress"
    role="progressbar"
  >
    <div
      class="mdc-linear-progress__buffer"
    >
      <div
        class="mdc-linear-progress__buffer-bar"
        style="flex-basis: 0%;"
      />
      <div
        class="mdc-linear-progress__buffer-dots"
      />
    </div>
    <div
      class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      style="transform: scaleX(0.5);"
    >
      <div
        class="mdc-linear-progress__bar-inner"
      />
    </div>
    <div
      class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
    >
      <div
        class="mdc-linear-progress__bar-inner"
      />
    </div>
  </nav>
</DocumentFragment>
`;
